lib/fuser_basic.h

```diff
@@ -26,10 +26,26 @@ namespace qsim {
 
 /**
  * Stateless object with methods for aggregating `Gate`s into `GateFused`.
+ * Measurement gates with equal times are fused together.
+ * User-defined controlled gates (controlled_by.size() > 0) and gates acting on
+ * more than two qubits are not fused.
+ * The template parameter Gate can be Gate type or a pointer to Gate type.
  */
 template <typename IO, typename Gate>
 struct BasicGateFuser final {
-  using GateFused = qsim::GateFused<Gate>;
+ private:
+  using RGate = typename std::remove_pointer<Gate>::type;
+
+  static const RGate& ConstRefToGate(const RGate& gate) {
+    return gate;
+  }
+
+  static const RGate& ConstRefToGate(const RGate* gate) {
```

Comment:
Should these be `GateToConstRef` instead? It looks like the output is a const ref, rather than the input.

Reply:
Replaced `ConstRefToGate` with `GateToConstRef`.

Reply:
Replaced `ConstRefToGate` with `GateToConstRef`.